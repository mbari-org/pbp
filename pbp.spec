# -*- mode: python ; coding: utf-8 -*-
"""
PyInstaller spec file for building standalone mbari-pbp distribution.

Usage:
    poetry run pyinstaller pbp.spec

This creates a directory bundle (--onedir) with faster startup than --onefile.
The dist/pbp/ directory contains the executable and all dependencies.
"""

import sys
from PyInstaller.utils.hooks import collect_data_files, collect_submodules

block_cipher = None

# Collect all submodules for packages that use dynamic imports
hiddenimports = [
    'pbp.hmb_gen.main_hmb_generator',
    'pbp.cloud.main_cloud',
    'pbp.hmb_plot.main_plot',
    'pbp.meta_gen.main_meta_generator',
]

# Add PyPAM and its dependencies
hiddenimports += collect_submodules('pypam')
hiddenimports += collect_submodules('soundfile')
hiddenimports += collect_submodules('h5netcdf')
hiddenimports += collect_submodules('netCDF4')
hiddenimports += collect_submodules('scipy')
hiddenimports += collect_submodules('numpy')

# Collect data files for packages that need them
datas = []
datas += collect_data_files('soundfile')
datas += collect_data_files('h5netcdf')
datas += collect_data_files('pypam')

a = Analysis(
    ['pbp/main_cli.py'],
    pathex=[],
    binaries=[],
    datas=datas,
    hiddenimports=hiddenimports,
    hookspath=[],
    hooksconfig={},
    runtime_hooks=[],
    excludes=[
        # Exclude test and development packages to reduce size
        'pytest',
        'mypy',
        'ruff',
        'IPython',
        'jupyter',
    ],
    win_no_prefer_redirects=False,
    win_private_assemblies=False,
    cipher=block_cipher,
    noarchive=False,
)

pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

exe = EXE(
    pyz,
    a.scripts,
    [],
    exclude_binaries=True,  # Changed: Create --onedir bundle
    name='pbp',
    debug=False,
    bootloader_ignore_signals=False,
    strip=False,
    upx=True,
    console=True,
    disable_windowed_traceback=False,
    argv_emulation=False,
    target_arch=None,
    codesign_identity=None,
    entitlements_file=None,
)

coll = COLLECT(
    exe,
    a.binaries,
    a.zipfiles,
    a.datas,
    strip=False,
    upx=True,
    upx_exclude=[],
    name='pbp',
)
