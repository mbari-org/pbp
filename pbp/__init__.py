import os

import xarray as xr

from pbp.logging_helper import PbpLogger


def save_dataset_to_netcdf(logger: PbpLogger, ds: xr.Dataset, filename: str) -> bool:
    logger.info(f"  - saving dataset to: {filename}")
    try:
        ds.to_netcdf(
            filename,
            engine="h5netcdf",
            encoding={
                "effort": {"_FillValue": None},
                "frequency": {"_FillValue": None},
                "sensitivity": {"_FillValue": None},
            },
        )
        return True
    except Exception as e:  # pylint: disable=broad-exception-caught
        error = f"Unable to save {filename}: {e}"
        logger.error(error)
        print(error)
        return False


def save_dataset_to_csv(logger: PbpLogger, ds: xr.Dataset, filename: str):
    logger.info(f"  - saving dataset to: {filename}")
    try:
        ds.to_pandas().to_csv(filename, float_format="%.1f")
    except Exception as e:  # pylint: disable=broad-exception-caught
        logger.error(f"Unable to save {filename}: {e}")


def get_cpus_to_use(num_cpus: int) -> int:
    cpu_count: int = os.cpu_count() or 1
    if num_cpus <= 0 or num_cpus > cpu_count:
        num_cpus = cpu_count
    return num_cpus


class PBPException(Exception):
    """
    Placeholder for a more specific exception.
    """

    def __init__(self, msg: str):
        super().__init__(f"PBPException({msg})")
