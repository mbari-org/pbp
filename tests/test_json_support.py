from typing import Any, List

import pytest
from pbp.json_support import get_intersecting_entries, parse_json_file
from pbp.logging_helper import create_logger


def _as_jsons(the_list: List[Any]) -> List[str]:
    return [e.to_json() for e in the_list]


@pytest.fixture
def json_entries():
    return list(parse_json_file("tests/json/2022/20220902.json"))


def test_json_parsing(json_entries, snapshot):
    assert _as_jsons(json_entries) == snapshot


def test_get_intersecting_entries(json_entries, snapshot):
    year, month, day = 2022, 9, 2

    logger = create_logger()

    def do_test(segment_size_in_mins: int, at_hour: int, at_minute: int):
        intersecting_entries = get_intersecting_entries(
            logger,
            json_entries,
            year,
            month,
            day,
            at_hour,
            at_minute,
            segment_size_in_mins=segment_size_in_mins,
        )
        assert _as_jsons(intersecting_entries) == snapshot(
            name=f"size={segment_size_in_mins:02} h={at_hour:02} m={at_minute:02}"
        )

    do_test(1, 0, 0)
    do_test(1, 23, 59)
    do_test(20, 0, 0)
    do_test(10, 23, 40)


@pytest.fixture
def json_entries_2():
    res = list(parse_json_file("tests/json/2022/20221102.json"))
    return res


def test_json_parsing_2(json_entries_2, snapshot):
    assert _as_jsons(json_entries_2) == snapshot
