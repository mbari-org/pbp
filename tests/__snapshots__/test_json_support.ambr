# serializer version: 1
# name: test_get_intersecting_entries[size=01 h=00 m=00]
  list([
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220901_235016.wav", "duration_secs": 600.0, "start": "2022-09-01T23:50:16.000Z"}, "start_secs": 584, "duration_secs": 16}',
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220902_000016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:00:16.000Z"}, "start_secs": 0, "duration_secs": 44}',
  ])
# ---
# name: test_get_intersecting_entries[size=01 h=23 m=59]
  list([
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220902_235016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:50:16.000Z"}, "start_secs": 524, "duration_secs": 60}',
  ])
# ---
# name: test_get_intersecting_entries[size=10 h=23 m=40]
  list([
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220902_233016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:30:16.000Z"}, "start_secs": 584, "duration_secs": 16}',
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220902_234016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:40:16.000Z"}, "start_secs": 0, "duration_secs": 584}',
  ])
# ---
# name: test_get_intersecting_entries[size=20 h=00 m=00]
  list([
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220901_235016.wav", "duration_secs": 600.0, "start": "2022-09-01T23:50:16.000Z"}, "start_secs": 584, "duration_secs": 16}',
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220902_000016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:00:16.000Z"}, "start_secs": 0, "duration_secs": 600}',
    '{"entry": {"uri": "file:///PAM_Archive/2022/09/MARS_20220902_001016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:10:16.000Z"}, "start_secs": 0, "duration_secs": 584}',
  ])
# ---
# name: test_json_parsing
  list([
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220901_235016.wav", "duration_secs": 600.0, "start": "2022-09-01T23:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_000016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_001016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_002016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_003016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_004016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_005016.wav", "duration_secs": 600.0, "start": "2022-09-02T00:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_010016.wav", "duration_secs": 600.0, "start": "2022-09-02T01:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_011016.wav", "duration_secs": 600.0, "start": "2022-09-02T01:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_012016.wav", "duration_secs": 600.0, "start": "2022-09-02T01:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_013016.wav", "duration_secs": 600.0, "start": "2022-09-02T01:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_014016.wav", "duration_secs": 600.0, "start": "2022-09-02T01:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_015016.wav", "duration_secs": 600.0, "start": "2022-09-02T01:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_020016.wav", "duration_secs": 600.0, "start": "2022-09-02T02:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_021016.wav", "duration_secs": 600.0, "start": "2022-09-02T02:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_022016.wav", "duration_secs": 600.0, "start": "2022-09-02T02:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_023016.wav", "duration_secs": 600.0, "start": "2022-09-02T02:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_024016.wav", "duration_secs": 600.0, "start": "2022-09-02T02:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_025016.wav", "duration_secs": 600.0, "start": "2022-09-02T02:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_030016.wav", "duration_secs": 600.0, "start": "2022-09-02T03:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_031016.wav", "duration_secs": 600.0, "start": "2022-09-02T03:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_032016.wav", "duration_secs": 600.0, "start": "2022-09-02T03:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_033016.wav", "duration_secs": 600.0, "start": "2022-09-02T03:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_034016.wav", "duration_secs": 600.0, "start": "2022-09-02T03:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_035016.wav", "duration_secs": 600.0, "start": "2022-09-02T03:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_040016.wav", "duration_secs": 600.0, "start": "2022-09-02T04:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_041016.wav", "duration_secs": 600.0, "start": "2022-09-02T04:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_042016.wav", "duration_secs": 600.0, "start": "2022-09-02T04:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_043016.wav", "duration_secs": 600.0, "start": "2022-09-02T04:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_044016.wav", "duration_secs": 600.0, "start": "2022-09-02T04:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_045016.wav", "duration_secs": 600.0, "start": "2022-09-02T04:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_050016.wav", "duration_secs": 600.0, "start": "2022-09-02T05:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_051016.wav", "duration_secs": 600.0, "start": "2022-09-02T05:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_052016.wav", "duration_secs": 600.0, "start": "2022-09-02T05:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_053016.wav", "duration_secs": 600.0, "start": "2022-09-02T05:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_054016.wav", "duration_secs": 600.0, "start": "2022-09-02T05:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_055016.wav", "duration_secs": 600.0, "start": "2022-09-02T05:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_060016.wav", "duration_secs": 600.0, "start": "2022-09-02T06:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_061016.wav", "duration_secs": 600.0, "start": "2022-09-02T06:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_062016.wav", "duration_secs": 600.0, "start": "2022-09-02T06:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_063016.wav", "duration_secs": 600.0, "start": "2022-09-02T06:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_064016.wav", "duration_secs": 600.0, "start": "2022-09-02T06:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_065016.wav", "duration_secs": 600.0, "start": "2022-09-02T06:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_070016.wav", "duration_secs": 600.0, "start": "2022-09-02T07:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_071016.wav", "duration_secs": 600.0, "start": "2022-09-02T07:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_072016.wav", "duration_secs": 600.0, "start": "2022-09-02T07:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_073016.wav", "duration_secs": 600.0, "start": "2022-09-02T07:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_074016.wav", "duration_secs": 600.0, "start": "2022-09-02T07:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_075016.wav", "duration_secs": 600.0, "start": "2022-09-02T07:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_080016.wav", "duration_secs": 600.0, "start": "2022-09-02T08:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_081016.wav", "duration_secs": 600.0, "start": "2022-09-02T08:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_082016.wav", "duration_secs": 600.0, "start": "2022-09-02T08:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_083016.wav", "duration_secs": 600.0, "start": "2022-09-02T08:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_084016.wav", "duration_secs": 600.0, "start": "2022-09-02T08:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_085016.wav", "duration_secs": 600.0, "start": "2022-09-02T08:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_090016.wav", "duration_secs": 600.0, "start": "2022-09-02T09:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_091016.wav", "duration_secs": 600.0, "start": "2022-09-02T09:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_092016.wav", "duration_secs": 600.0, "start": "2022-09-02T09:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_093016.wav", "duration_secs": 600.0, "start": "2022-09-02T09:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_094016.wav", "duration_secs": 600.0, "start": "2022-09-02T09:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_095016.wav", "duration_secs": 600.0, "start": "2022-09-02T09:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_100016.wav", "duration_secs": 600.0, "start": "2022-09-02T10:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_101016.wav", "duration_secs": 600.0, "start": "2022-09-02T10:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_102016.wav", "duration_secs": 600.0, "start": "2022-09-02T10:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_103016.wav", "duration_secs": 600.0, "start": "2022-09-02T10:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_104016.wav", "duration_secs": 600.0, "start": "2022-09-02T10:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_105016.wav", "duration_secs": 600.0, "start": "2022-09-02T10:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_110016.wav", "duration_secs": 600.0, "start": "2022-09-02T11:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_111016.wav", "duration_secs": 600.0, "start": "2022-09-02T11:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_112016.wav", "duration_secs": 600.0, "start": "2022-09-02T11:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_113016.wav", "duration_secs": 600.0, "start": "2022-09-02T11:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_114016.wav", "duration_secs": 600.0, "start": "2022-09-02T11:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_115016.wav", "duration_secs": 600.0, "start": "2022-09-02T11:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_120016.wav", "duration_secs": 600.0, "start": "2022-09-02T12:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_121016.wav", "duration_secs": 600.0, "start": "2022-09-02T12:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_122016.wav", "duration_secs": 600.0, "start": "2022-09-02T12:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_123016.wav", "duration_secs": 600.0, "start": "2022-09-02T12:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_124016.wav", "duration_secs": 600.0, "start": "2022-09-02T12:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_125016.wav", "duration_secs": 600.0, "start": "2022-09-02T12:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_130016.wav", "duration_secs": 600.0, "start": "2022-09-02T13:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_131016.wav", "duration_secs": 600.0, "start": "2022-09-02T13:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_132016.wav", "duration_secs": 600.0, "start": "2022-09-02T13:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_133016.wav", "duration_secs": 600.0, "start": "2022-09-02T13:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_134016.wav", "duration_secs": 600.0, "start": "2022-09-02T13:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_135016.wav", "duration_secs": 600.0, "start": "2022-09-02T13:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_140016.wav", "duration_secs": 600.0, "start": "2022-09-02T14:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_141016.wav", "duration_secs": 600.0, "start": "2022-09-02T14:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_142016.wav", "duration_secs": 600.0, "start": "2022-09-02T14:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_143016.wav", "duration_secs": 600.0, "start": "2022-09-02T14:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_144016.wav", "duration_secs": 600.0, "start": "2022-09-02T14:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_145016.wav", "duration_secs": 600.0, "start": "2022-09-02T14:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_150016.wav", "duration_secs": 600.0, "start": "2022-09-02T15:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_151016.wav", "duration_secs": 600.0, "start": "2022-09-02T15:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_152016.wav", "duration_secs": 600.0, "start": "2022-09-02T15:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_153016.wav", "duration_secs": 600.0, "start": "2022-09-02T15:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_154016.wav", "duration_secs": 600.0, "start": "2022-09-02T15:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_155016.wav", "duration_secs": 600.0, "start": "2022-09-02T15:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_160016.wav", "duration_secs": 600.0, "start": "2022-09-02T16:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_161016.wav", "duration_secs": 600.0, "start": "2022-09-02T16:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_162016.wav", "duration_secs": 600.0, "start": "2022-09-02T16:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_163016.wav", "duration_secs": 600.0, "start": "2022-09-02T16:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_164016.wav", "duration_secs": 600.0, "start": "2022-09-02T16:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_165016.wav", "duration_secs": 600.0, "start": "2022-09-02T16:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_170016.wav", "duration_secs": 600.0, "start": "2022-09-02T17:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_171016.wav", "duration_secs": 600.0, "start": "2022-09-02T17:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_172016.wav", "duration_secs": 600.0, "start": "2022-09-02T17:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_173016.wav", "duration_secs": 600.0, "start": "2022-09-02T17:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_174016.wav", "duration_secs": 600.0, "start": "2022-09-02T17:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_175016.wav", "duration_secs": 600.0, "start": "2022-09-02T17:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_180016.wav", "duration_secs": 600.0, "start": "2022-09-02T18:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_181016.wav", "duration_secs": 600.0, "start": "2022-09-02T18:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_182016.wav", "duration_secs": 600.0, "start": "2022-09-02T18:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_183016.wav", "duration_secs": 600.0, "start": "2022-09-02T18:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_184016.wav", "duration_secs": 600.0, "start": "2022-09-02T18:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_185016.wav", "duration_secs": 600.0, "start": "2022-09-02T18:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_190016.wav", "duration_secs": 600.0, "start": "2022-09-02T19:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_191016.wav", "duration_secs": 600.0, "start": "2022-09-02T19:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_192016.wav", "duration_secs": 600.0, "start": "2022-09-02T19:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_193016.wav", "duration_secs": 600.0, "start": "2022-09-02T19:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_194016.wav", "duration_secs": 600.0, "start": "2022-09-02T19:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_195016.wav", "duration_secs": 600.0, "start": "2022-09-02T19:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_200016.wav", "duration_secs": 600.0, "start": "2022-09-02T20:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_201016.wav", "duration_secs": 600.0, "start": "2022-09-02T20:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_202016.wav", "duration_secs": 600.0, "start": "2022-09-02T20:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_203016.wav", "duration_secs": 600.0, "start": "2022-09-02T20:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_204016.wav", "duration_secs": 600.0, "start": "2022-09-02T20:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_205016.wav", "duration_secs": 600.0, "start": "2022-09-02T20:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_210016.wav", "duration_secs": 600.0, "start": "2022-09-02T21:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_211016.wav", "duration_secs": 600.0, "start": "2022-09-02T21:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_212016.wav", "duration_secs": 600.0, "start": "2022-09-02T21:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_213016.wav", "duration_secs": 600.0, "start": "2022-09-02T21:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_214016.wav", "duration_secs": 600.0, "start": "2022-09-02T21:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_215016.wav", "duration_secs": 600.0, "start": "2022-09-02T21:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_220016.wav", "duration_secs": 600.0, "start": "2022-09-02T22:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_221016.wav", "duration_secs": 600.0, "start": "2022-09-02T22:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_222016.wav", "duration_secs": 600.0, "start": "2022-09-02T22:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_223016.wav", "duration_secs": 600.0, "start": "2022-09-02T22:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_224016.wav", "duration_secs": 600.0, "start": "2022-09-02T22:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_225016.wav", "duration_secs": 600.0, "start": "2022-09-02T22:50:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_230016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:00:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_231016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:10:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_232016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:20:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_233016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:30:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_234016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:40:16.000Z"}',
    '{"uri": "file:///PAM_Archive/2022/09/MARS_20220902_235016.wav", "duration_secs": 600.0, "start": "2022-09-02T23:50:16.000Z"}',
  ])
# ---
# name: test_json_parsing_2
  list([
    '{"uri": "file://test.wav", "duration_secs": 60.0, "start": "2022-11-02T00:00:00.000Z"}',
  ])
# ---
